pkgs/games/sgfutils/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, openssl
+, iconv
+, makeWrapper
+, imagemagick
+, makeFontsConf
+}:
+stdenv.mkDerivation
+{
+  pname = "sgfutils";
+  version = "unstable-2017-11-27";
+  buildInputs = [ openssl makeWrapper ] ++ lib.optionals stdenv.isDarwin [ iconv ];
+  src = fetchFromGitHub {
+    owner = "yangboz";
+    repo = "sgfutils";
+    rev = "11ab171c46cc16cc71ac6fc901d38ea88d6532a4";
+    hash = "sha256-KWYgTxz32WK3MKouj1WAJtZmleKt5giCpzQPwfWruZQ=";
+  };
+  buildPhase = "make all";
+  installPhase = ''
+    mkdir -p $out/bin
+    cp sgf sgfsplit sgfvarsplit sgfstrip sgfinfo sgfmerge sgftf \
+      sgfcheck sgfdb sgfdbinfo sgfcharset sgfcmp sgfx \
+      ngf2sgf nip2sgf nk2sgf gib2sgf sgftopng ugi2sgf \
+      $out/bin
+  '';
+  postFixup = ''
+    wrapProgram $out/bin/sgftopng \
+      --set PATH ${lib.makeBinPath [ imagemagick ]} \
+      --set-default FONTCONFIG_FILE ${ makeFontsConf { fontDirectories = []; } }
```

Comment:
Wouldn't it be better to patch the source file to specify the full path of `convert` instead of wrapping?

Reply:
it wouldn't IMO. patches are more fragile - if the source ever changes, which it admittedly might not, then it becomes a chore to change the patch as well.

Reply:
I'm talking about substituteInPlace, you don't have to change that unless they change the binary name. Obviously a diff patch is unreasonable, especially because you'd have to generate it from a template first.

Reply:
The same point stands with substituteInPlace. Why define a brittle patch when a robust wrapper gets the same thing done?

Reply:
I suppose so, I just think that's generally preferable over the env leakage and messed up argv[0] caused by wrappers, which can both cause issues in other packages down the line. The Qt wrapper has bitten me just a few days ago by Konsole's (or something else?) setting QT_PLUGIN_PATH which caused me to miss some dependencies of something I was packaging because they were already in the inherited environment, and I only caught it because I tested the package on my Mac afterwards.

Though tbf in reality both are unlikely to cause a big problem here other than an ugly name in the help output (if even) and the process list.

Reply:
anyway I just realized that even if `convert` was substituted in place, the FONTCONFIG_FILE env var should be set.

Reply:
Ah yeah, that's true.